File: app/src/main/java/williamroocka/tut/onboard/MyApplication.java
package williamroocka.tut.onboard;

import android.app.Application;
import android.support.multidex.MultiDex;

import com.android.volley.RequestQueue;

import williamroocka.tut.onboard.network.VolleySingleton;


public class MyApplication extends Application {
    private RequestQueue requestQueue;
    private static MyApplication mInstance;

    @Override
    public void onCreate() {
        super.onCreate();
        mInstance = this;
        requestQueue = VolleySingleton.getInstance(getApplicationContext()).getRequestQueue();
        MultiDex.install(this);
    }

    public static synchronized MyApplication getInstance() {
        return mInstance;
    }

    public RequestQueue getVolleyRequestQueue() {
        return requestQueue;
    }
}

File: app/src/main/java/williamroocka/tut/onboard/MyApplication.kt
package williamroocka.tut.onboard

import android.app.Application
import android.support.multidex.MultiDex

import com.android.volley.RequestQueue

import williamroocka.tut.onboard.network.VolleySingleton

open class MyApplication : Application() {
    companion object {
        lateinit var requestQueue: RequestQueue
        lateinit var mInstance: MyApplication
    }

    override fun onCreate() {
        super.onCreate()
        mInstance = this
        requestQueue = VolleySingleton.getInstance(getInstance()).getRequestQueue()
        MultiDex.install(this)
    }

    fun getInstance(): MyApplication {
        synchronized(this) {
            return mInstance
        }
    }

    fun getVolleyRequestQueue(): RequestQueue = requestQueue

}


